pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -194,6 +194,43 @@ func installServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
+// removeCaBundleFromApiservice removes spec.caBundle from v1beta1.custom.metrics.k8s.io.
+// This is necessary as `insecureSkipTLSVerify: true` in resources/knative-serving-x.y.z.yaml conflicts
+// with service.alpha.openshift.io/inject-cabundle annotaiton in addCaBundleToApiservice.
+func removeCaBundleFromApiservice(instance *servingv1alpha1.KnativeServing) error {
```

Comment:
I know that this is ugly hack. But applying `resources/knative-serving-0.7.1.yaml` second time (= when operator restarts) produces following error :crying_cat_face:

```
"error":"APIService.apiregistration.k8s.io \"v1beta1.custom.metrics.k8s.io\" is invalid: spec.insecureSkipTLSVerify: Invalid value: true: may not be true if caBundle is present"
```

@jcrossley3  @markusthoemmes @maschmid Please let me know if  you have any better idea.

Reply:
Can we not transform the existing APIService programatically like we do below with some deployments?